ui/addon/src/main/java/org/jboss/windup/ui/RulesetUpdateChecker.java

```diff
@@ -0,0 +1,111 @@
+package org.jboss.windup.ui;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import javax.enterprise.event.Observes;
+import javax.inject.Singleton;
+
+import org.jboss.forge.addon.dependencies.Coordinate;
+import org.jboss.forge.addon.dependencies.DependencyResolver;
+import org.jboss.forge.addon.dependencies.builder.CoordinateBuilder;
+import org.jboss.forge.addon.dependencies.builder.DependencyQueryBuilder;
+import org.jboss.forge.furnace.Furnace;
+import org.jboss.forge.furnace.event.PostStartup;
+import org.jboss.forge.furnace.services.Imported;
+import org.jboss.windup.config.furnace.FurnaceHolder;
+import org.jboss.windup.util.PathUtil;
+
+@Singleton
+public class RulesetUpdateChecker
+{
+
+    private static final String META_INF_VERSION_KEY = "version=";
+    public static final String RULESET_CORE_DIRECTORY = "migration-core";
+
+    public void perform(@Observes PostStartup event)
+    {
+        boolean isDependencies = event.getAddon().getId().toString().contains("org.jboss.windup.ui:windup-ui");
+        if(isDependencies) {
+            Furnace furnace = FurnaceHolder.getFurnace();
```

Comment:
I believe you can @Inject furnace instead of doing this. You are, after all, in a Furnace CDI environment at this point in the system. I'd switch to useing @Inject, or add Furnace to the method parameter list for another style of injection.